Slack/SlackMessaging.cs

```diff
@@ -1,31 +1,27 @@
 using System;
-using System.Linq;
-using System.Reactive;
 using System.Reactive.Linq;
-using System.Reactive.Subjects;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using RestEase;
 
 namespace Slack.Json.Slack
 {
-
     public class SlackMessaging : ISlackMessaging
     {
         private readonly AppOptions options;
-        private readonly ISubject<(string channel, SlackMessageModel model)> messageSubject = Subject.Synchronize(new Subject<(string channel, SlackMessageModel model)>());
-        private readonly TimeSpan groupingTimeForMessages = TimeSpan.FromSeconds(15);
         private readonly string user = "GitHub";
+        private readonly MessageThrottler throttler;
 
-        public SlackMessaging(IOptions<AppOptions> options, ILogger<SlackMessaging> logger)
+        public SlackMessaging(IOptions<AppOptions> options, ILogger<SlackMessaging> logger, MessageThrottler throttler)
         {
             this.options = options.Value;
+            this.throttler = throttler;
 
             if (string.IsNullOrEmpty(this.options.SlackIntegrationUri))
                 throw new ArgumentException(nameof(this.options.SlackIntegrationUri));
 
-            ThrottleByTittle(this.messageSubject.Synchronize())
+            throttler.Messages()
                 .Subscribe(async toSend =>
                 {
                     try
```

Comment:
This is about row 37:

Consider using 
`logger.LogError(ex, $"Error occurred while tried to send message '{toSend.model.Title}' to channel '{toSend.channel}', error: {ex}");` so if we change logging settings we can actually use the exception object with logger.

Other thing to consider (more FYI thing with Application Insights, even if it is not used here) is to let the logger do string formatting (as opposed to string interpolation. This way logging parameters are automatically added as custom parameters in application insights an it's easier to search logs.

For example, following logging version would produce logging items where we can easily filter explicitly with the title, instead of some "message contains #test-channel" or similar scraping solution.
`logger.LogError(ex, "Error occurred while tried to send message '{title}' to channel '{channel}', error: {ex}",toSend.model.Title, toSend.channel, ex);`